### PR TITLE
Add to-do list widget

### PR DESCRIPTION
**Pull request by massimocode**

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -1,28 +1,45 @@
...
+	padding-left: 2px;
+	padding-right: 2px;
+}
+
+
+/*--END OF CALCULATOR*--------------------------------/
+
+/*TO-DO-LIST*/
+
+.task-tex{
```
Comment (massimocode) [inline style.css]: What is this CSS rule for?

File: style.css [hunk 2]
```diff
@@ -1,28 +1,45 @@
...
+padding-top: 15px;
+}
+.task-text button, .task-tex input{
```
Comment (massimocode) [inline style.css]: There is no element on the screen which matches the selector `.task-tex input`

File: spec/todo-list.spec.js [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+            expect(instance.isMarkNotDoneVisible()).to.equal(false);
+          });
+
+          it('The delete button should be displayed', () => {
+            expect(instance.isDeleteVisible()).to.equal(true);
+          });
+        });
+      });
+
+      // Test Driven Development
```
Comment (massimocode) [inline spec/todo-list.spec.js]: All of these comments can be deleted

File: calculator.html [hunk 4]
```diff
@@ -5,54 +5,94 @@
...
+            <div>
+              <ul class="list-group" data-bind="foreach: tasks">
+                <li class="list-group-item"><label><input type="checkbox" data-bind="checked: isSelected">
+                  <span data-bind="text: name, style: { color: isDone() ? 'green' : 'black' }">  </span></label>
+                </li>
+              </ul>
+
+            </div>
+            <div>
+              <button class="btn btn-success add-margin-btn" data-bind="click: markAsDone, visible: isMarkAsDoneVisible()">Make as done</button>
```
Comment (massimocode) [inline calculator.html]: Should say "Mark as done"

File: calculator.html [hunk 5]
```diff
@@ -5,54 +5,94 @@
...
+              <button class="btn btn-success add-margin-btn" data-bind="click: markNotDone, visible: isMarkNotDoneVisible()">Make as not done</button>
```
Comment (massimocode) [inline calculator.html]: Should say "Mark as not done"

File: calculator.html [hunk 6]
```diff
@@ -5,54 +5,94 @@
...
+              <button class="btn btn-danger" data-bind="click: deleteTasks, visible: isDeleteVisible()">Delete</button>
+            </div>
+            <div class="task-text">
+              <form class="form-inline">
+                <div class="form-group">
+                  <input type="text" placeholder="Enter something to do.." class="form-control" data-bind="value: newTaskText">
```
Comment (massimocode) [inline calculator.html]: Enter something to do... (ellipsis is typically 3 dots not 2 https://en.wikipedia.org/wiki/Ellipsis)

File: calculator.html [hunk 7]
```diff
@@ -5,54 +5,94 @@
...
-            <button class="btn btn-block" data-bind="click: clear">C</button>
-            <button class="btn btn-block" data-bind="click: addPlus">+</button>
-            <button class="btn btn-block" data-bind="click: addMinus">-</button>
-            <button class="btn btn-block" data-bind="click: equal">=</button>
+          <div class="panel-body">
+            <h4 data-bind="visible: !tasks().length">You have nothing to do, great!</h4>
+            <div>
+              <ul class="list-group" data-bind="foreach: tasks">
+                <li class="list-group-item"><label><input type="checkbox" data-bind="checked: isSelected">
+                  <span data-bind="text: name, style: { color: isDone() ? 'green' : 'black' }">  </span></label>
```
Comment (massimocode) [inline calculator.html]: Try using the css binding instead with `'text-success': isDone`

File: calculator.html [hunk 8]
```diff
@@ -5,54 +5,94 @@
...
+              </div>
+              <div class="col-xs-3">
+                <button class="btn btn-block" data-bind="click: clear">C</button>
+                <button class="btn btn-block" data-bind="click: addPlus">+</button>
+                <button class="btn btn-block" data-bind="click: addMinus">-</button>
+                <button class="btn btn-block" data-bind="click: equal">=</button>
+              </div>
+            </div>
+            <button class="btn btn-block btn-danger remove-button" data-bind="click: $parent.removeCalculator, visible: $parent.calculators().length>
+        1">Remove</button>
```
Comment (massimocode) [inline calculator.html]: This should probably be joined with the previous line. It's not very good practice to split a tag over 2 lines if there's not a very good reason to do so.

File: lib/todo-list.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        window.ToDoList = factory(ko);
+    }
+} (function (ko) {
+    return function ToDoList() {
+        'use strict';
+        let tasks = ko.observableArray([]);
+        let newTaskText = ko.observable("");
+
+        function addTask() {
+            if (newTaskText() === "") return;
```
Comment (massimocode) [inline lib/todo-list.js]: I've noticed this line of code is new. It makes sense that when you click the Add Task button, an empty task should not be added. But I think we should add a unit test for this. Can you do that please?

File: style.css (hunk 2 above)
Comment (SMH110) [inline style.css]: it's matches this 
```
<input type="text" placeholder="Enter something to do.." class="form-control" data-bind="value: newTaskText">
                  <button class="btn btn-default" type="submit" data-bind="click: addTask">Add</button>
```

**Pull request merged**